.github/workflows/reusable-release.yml

```diff
@@ -289,20 +331,62 @@ jobs:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      GH_REPO: ${{ github.repository }}
+      GITHUB_REPO_URL: ${{ github.server_url }}/${{ github.repository }}
     steps:
+      - name: Common values
+        run: |
+          GITHUB_ACTIONS_PATH="$GITHUB_REPO_URL/actions"
+          GITHUB_WORKFLOW_FILE="release.yml" # Could be derived from $GITHUB_WORKFLOW_REF
+          GITHUB_WORKFLOW_URL="$GITHUB_ACTIONS_PATH/workflows/$GITHUB_WORKFLOW_FILE"
+          
+          cat << EndOfFile >> $GITHUB_ENV
+          GITHUB_WORKFLOW_FILE=$GITHUB_WORKFLOW_FILE
+          GITHUB_WORKFLOW_LINK=[GitHub UI]($GITHUB_WORKFLOW_URL)
+          GITHUB_WORKFLOW_RUN_LINK=[#${{ github.run_number }}]($GITHUB_ACTIONS_PATH/runs/${{ github.run_id }})
+          EndOfFile
       - name: Create Github Release
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          GH_REPO: ${{ github.repository }}
+        if: needs.init.outputs.release_type == 'FULL_MAIN_BRANCH'
         run: |
-          gh release create $RELEASE_TAG --verify-tag --generate-notes --notes "Release run: [#${{ github.run_number }}](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})"
-      - name: Job summary
+          gh release create $RELEASE_TAG --verify-tag --generate-notes --notes "Release run: $GITHUB_WORKFLOW_RUN_LINK"
+          echo "GitHub Release notes: [$RELEASE_TAG]($GITHUB_REPO_URL/releases/tag/$RELEASE_TAG)" >> $GITHUB_STEP_SUMMARY
+      - name: Update PR with comment
+        if: needs.init.outputs.release_type == 'PREVIEW_FEATURE_BRANCH'
         run: |
-          echo "GitHub Release notes: [$RELEASE_TAG](${{ github.server_url }}/${{ github.repository }}/releases/tag/$RELEASE_TAG)" >> $GITHUB_STEP_SUMMARY
+          cat << EndOfFile > comment_body.txt
+          @${{github.actor}} has published a preview version of this PR with release workflow run $GITHUB_WORKFLOW_RUN_LINK, based on commit ${{ github.sha }}:
+          
+          $RELEASE_VERSION
+          
+          <details>
+          <summary>Want to make another preview release?</summary>
+          
+          Click 'Run workflow' in the $GITHUB_WORKFLOW_LINK, specifying the $GITHUB_REF_NAME branch, or use the [GitHub CLI](https://cli.github.com/) command:
+          
+          gh workflow run $GITHUB_WORKFLOW_FILE --ref $GITHUB_REF_NAME
+          
+          </details>
+          
+          <details>
+          <summary>Want to make a full release after this PR is merged?</summary>
+          
+          Click 'Run workflow' in the $GITHUB_WORKFLOW_LINK, leaving the branch as the default, or use the [GitHub CLI](https://cli.github.com/) command:
+          
+          gh workflow run $GITHUB_WORKFLOW_FILE
+          
+          </details>
+          EndOfFile
+          
+          cat comment_body.txt
+          
+          gh pr comment ${{ github.ref_name }} --body-file comment_body.txt >> $GITHUB_STEP_SUMMARY
```

Comment:
[Problems with Bash interpreting backticks](https://github.com/rtyley/sample-project-using-gha-scala-library-release-workflow/actions/runs/7399435634/job/20130944058) have meant that the comment message isn't quite as pretty as I'd like - ideally there would be backticks for [styling code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks), like [this](https://github.com/rtyley/sample-project-using-gha-scala-library-release-workflow/pull/1#issuecomment-1872955682):

![image](https://github.com/guardian/gha-scala-library-release-workflow/assets/52038/285f2b1e-ddd3-4eb8-803f-7ddbde1ca007)

For the time being this PR avoids them in the generated PR comment.